master/internal/resourcemanagers/resource_manager_config.go

```diff
@@ -0,0 +1,124 @@
+package resourcemanagers
+
+import (
+	"encoding/json"
+
+	"github.com/pkg/errors"
+
+	"github.com/determined-ai/determined/master/internal/provisioner"
+	"github.com/determined-ai/determined/master/pkg/check"
+	"github.com/determined-ai/determined/master/pkg/union"
+)
+
+// ResolveConfig applies backwards compatibility for the old scheduler
+// and provisioner configuration.
+func ResolveConfig(
+	sConf *Config,
+	pConf *provisioner.Config,
```

Comment:
non-blocking: can we use full names here e.g, (schedulerConf, provisionerConf) find the code hard to read with these acronyms.